### DISCUSSION TITLE
How to debug with IntelliJ/CLion?

### DISCUSSION THREAD

**Discussion by vjpr** [napi-rs/napi-rs#722] [Q&A] [answered]
I am wondering if there is a way to step-through the native Rust code from IntelliJ?

With cmake and gyp extensions written in C++/Objective-C I can usually debug with CLion/AppCode.

Related: https://github.com/intellij-rust/intellij-rust/issues/7763

**Comment by Brooooooklyn** [comment]
Is that article help https://rustc-dev-guide.rust-lang.org/debugging-support-in-rustc.html#debuggers ?
I'm not familiar with IntelliJ, so I may not be able to help you

**Comment by forehalo** [comment]
BTW, I know the way step into rust code in VSCode by using [vscode-lldb](https://github.com/vadimcn/vscode-lldb) plugin.

```jsonm
// .vscode/launch.json
{
  configurations: [{
      "type": "lldb",
      "request": "launch",
      "sourceLanguages": ["rust"],
      "name": "Debug example",
      "program": "node",
      "args": ["${workspaceFolder}/test.js"],
      "cwd": "${workspaceFolder}"
  }]
}
```
```js
// test.js
const { test } = require('path/to/addon.node')

test()
```

You would go stepped into native code successfully.

![image](https://user-images.githubusercontent.com/8281226/131643723-bdb907ba-1773-4945-8657-e2e23d260cd5.png)

**Reply by daydayhappychao** [reply]
Didn't work for me

**Comment by vjpr** [comment, marked as answer]
So I was trying to get it to work with the SWC project. I created a new `napi-rs` project from template, and in CLion it works perfectly. Repro here: https://github.com/vjpr/issue-napi-rs-intellij

- New run configuration
- Custom Build Application
- Exe: path/to/node
- Set working dir
- Pass your JS file as param
- Place breakpoint in `src/lib.rs`
- Click debug for this run config
- Should work

So I'm thinking its something to do with how SWC is setup or how I am running things.

**Reply by zhusjfaker** [reply]
i  have same issue, is there any good way to solve this issue
